specs/scenarios/telemetry.md

```diff
@@ -0,0 +1,93 @@
+- Adoption of the different platforms we offer support for.
+
+### Privacy
+
+We should be very careful about the information we collect. Before rolling out this feature, we should have full agreement with CELA on our approach, the way we collect and protect the data. In general:
+
+- We should offer a way to opt-out of the telemetry collection.
+- We should not collect any information that could be considered sensitive. 
+- We should not collect any information that could be used to identify a person, an application, or an API. 
+- We should not collect any information during the runtime of the application.
+
+### Examples of questions we want to answer
+
+- How many users are using the CLI?
+- How many users are using the CLI via the extension?
+- What is the current growth trajectory of the usage of kiota?
+- What are the most used commands?
+- How are the commands used? Which parameters are being used?
+- What are the most used languages?
+- What are the most used platforms?
+- Is there a difference in how the CLI is being used between platforms?
+- Do we have users using old versions of kiota? Why?
+- When launching new capabilities, what is the adoption rate? How long does it take for users to adopt the new capabilities?
+- Can we identify a spike in error being returned by Kiota? What is the impact of the error? What command generates the error? What parameters are being used?
+
+### Data collected
+
+#### Basic data being collected
+
+For every command, we should collect the following information:
+
+- Timestamp
+- Operating system
+- Operating system version
+- Source (CLI or extension)
+- Acquisition channel (dotnet tool, binaries, homebrew, asdf, extension, etc.)
+- Kiota version
+- VS Code extension version (if applicable)
+- Command name
+- Command parameters being used (without their values)
+- Command execution time
+- Command result (success or failure)
+
+#### Command-specific data being collected
+
+Every command has a different set of parameters. We should collect relevant parameters (and their values) for each command. The data collected shouldn't include any information that could be considered sensitive, only system-related information. Each command specification should include the list of parameters that should be collected and whether their values should be collected or not.
+
+The list of commands and their parameters can be found in the [CLI Commands](../cli/index.md) section. Each parameter indicates whether its value should be collected or not.
+
+### Opting-out
+
+We should offer a way to opt-out of the telemetry collection. This should be done in a very similar way that the `dotnet` CLI does (https://learn.microsoft.com/en-us/dotnet/core/tools/telemetry). To opt out of the telemetry feature, set the KIOTA_CLI_TELEMETRY_OPTOUT environment variable to 1 or true.
+
+Every time the CLI is installed and updated, we should inform the user about the telemetry feature and how to opt-out of it.
```

Comment:
Can we know whether they have already opt-out to not keep showing this message if they already have disabled telemetry?

Reply:
Fair. Updating the wording.